api/src/repositories/submission-repository.ts

```diff
@@ -213,6 +197,39 @@ export class SubmissionRepository extends BaseRepository {
     return response.rows[0];
   }
 
+  /**
+   * Insert a new submission record, returning the record having the matching UUID if it already exists
+   *
+   * @param {IInsertSubmissionRecord} submissionData
+   * @return {*}  {Promise<{ submission_id: number }>}
+   * @memberof SubmissionRepository
+   */
+  async getOrInsertSubmissionRecord(submissionData: IInsertSubmissionRecord): Promise<{ submission_id: number }> {
+    const sqlStatement = SQL`
+      INSERT INTO submission (
+        source_transform_id,
+        uuid
+      ) VALUES (
+        ${submissionData.source_transform_id},
+        ${submissionData.uuid}
+      )
+      ON CONFLICT (uuid) DO UPDATE SET uuid = ${submissionData.uuid}
```

Comment:
I'll add a comment explaining this, but basically if you put `ON CONFLICT ... DO NOTHING`, then `RETURNING` will fail to return anything